hedyorg/hedy
# [FIX]: Fix removing numbers from lists

**Description**

Added a condition to the if statement on line 1612 as strings that are numeric are not handled correctly by this function. Currently, strings that are numeric can not be properly removed from the list_var that holds them in the underlying implementation, so the remove function is ineffective for arrays of strings that hold numbers. These string types that are numeric must be treated as any other string to allow for the remove function to work as intended 


**Fixes _issue or discussion number_**

https://github.com/hedyorg/hedy/issues/3799

**How to test**

Use example from bug report and confirm it returns “list index out of range”. Afterwards add “add 4 to ff” above print statement on levels 4-11 like so- 

ff is 1, 2
remove 1 from ff
remove 2 from ff
Add 4
print ff at random

It should print 4 every time when you run because that is the only number stored afterwards. 


**Checklist**
Done? Check if you have it all in place using this list:*
  

If you're unsure about any of these, don't hesitate to ask. We're here to help!
